jenkins/bootstrap.py

```diff
@@ -735,11 +737,16 @@ def job_args(args):
     return [os.path.expandvars(a) for a in args]
 
 
-def job_script(job, use_json):
+def job_script(job, use_json, jobs_dir):
     """Return path to script for job."""
+    jobs_dir_resolver = lambda f: test_infra('jobs/%s' % f)
+
+    if jobs_dir:
+        jobs_dir_resolver = lambda f: os.path.join(jobs_dir, f)
+
     if not use_json:
-        return [test_infra('jobs/%s.sh' % job)]
-    with open(test_infra('jobs/config.json')) as fp:
+        return [jobs_dir_resolver('%s.sh' % job)]
+    with open(jobs_dir_resolver('config.json')) as fp:
         config = json.loads(fp.read())
     job_config = config[job]
     cmd = test_infra('scenarios/%s.py' % job_config['scenario'])
```

Comment:
maybe create a `scenario_resolver` as well, so that you can point your scenario to customized path, rather than always 'root/scenarios/`